bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/ICalendarHandler.java

```diff
@@ -154,14 +152,19 @@ public void initialize() {
             }
             final long refreshTime = refreshTimeBD.longValue();
             if (calendarFile.isFile()) {
-                if (reloadCalendar()) {
-                    updateStatus(ThingStatus.ONLINE);
-                    updateStates();
-                    rescheduleCalendarStateUpdate();
-                } else {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                            "The calendar seems to be configured correctly, but the local copy of calendar could not be loaded.");
-                }
+                updateStatus(ThingStatus.UNKNOWN);
+
+                scheduler.submit(() -> {
+                    // reload calendar file asynchronously
+                    if (reloadCalendar()) {
+                        updateStatus(ThingStatus.ONLINE);
+                        updateStates();
+                        rescheduleCalendarStateUpdate();
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "The calendar seems to be configured correctly, but the local copy of calendar could not be loaded.");
+                    }
+                });
                 pullJobFuture = scheduler.scheduleWithFixedDelay(regularPull, refreshTime, refreshTime,
                         TimeUnit.MINUTES);
```

Comment:
Maybe it would be better to just let the scheduled pull task handle the asynchronous part instead?
```suggestion
                updateStatus(ThingStatus.UNKNOWN);
                pullJobFuture = scheduler.scheduleWithFixedDelay(regularPull, 0, refreshTime,
                        TimeUnit.MINUTES);
```

Reply:
I am not sure if that will work. IIRC downloading the calendar is not part of the pull job. We might want to ask @daMihe for his opinion.

Reply:
Asynchronously loading the calendar works, it's already triggered by `PullJob` via `CalendarUpdateListener.onCalendarUpdated()` this way (regularly via `scheduleWithFixedDelay`). However, just using the `PullJob` for reloading may cause the calendar to be in `OFFLINE` state if e.g. being offline even if a cached copy exists. So i think this is not a good idea.

Reply:
Well this PR will put the calendar into an UNKNOWN state, which is a kind of offline state, while it loads the cached copy anyway. If we can assert that any stored copy of the calendar is a valid one then I can see allowing the binding to initially enter an ONLINE state if it fined a file is present already.

Reply:
Yes, sounds reasonable. I just saw that `reloadCalendar()` calls `rescheduleCalendarStateUpdate()` too which makes the second call inside the `if` superfluous. I will remove that too.

Reply:
Yes this seems to be redundant. You can also remove the `updateStatus` inside the if as its already done in `updateStates`. I assume this went into code about time and after several changes.

Reply:
Dine in https://github.com/openhab/openhab-addons/pull/9227/commits/906c1c273faecf1a243199598b2b832554b4627d.